validator/client/streaming/runner.go

```diff
@@ -0,0 +1,153 @@
+package streaming
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/featureconfig"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"go.opencensus.io/trace"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Validator interface defines the primary methods of a validator client.
+type Validator interface {
+	Done()
+	WaitForChainStart(ctx context.Context) error
+	WaitForSync(ctx context.Context) error
```

Comment:
Why two different methods for sync ?

Reply:
That's always been there. It's gated behind a feature flag that Ivan created to simplify the validator runtime but we've never enabled it, so both code paths still exist